java/code/src/com/suse/oval/OsFamily.java

```diff
@@ -15,25 +15,86 @@
 
 package com.suse.oval;
 
+import com.redhat.rhn.domain.server.Server;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * This enum defines the operating system families for which we can retrieve OVAL data.
+ * */
 public enum OsFamily {
-    LEAP("openSUSE Leap", "leap", "opensuse"),
-    SUSE_LINUX_ENTERPRISE_SERVER("SUSE Linux Enterprise Server", "sles", "suse"),
-    SUSE_LINUX_ENTERPRISE_DESKTOP("SUSE Linux Enterprise Desktop", "sled", "suse"),
-    REDHAT_ENTERPRISE_LINUX("Red Hat Enterprise Linux", "enterprise_linux", "redhat"),
-    UBUNTU("Ubuntu", "ubuntu", "canonical"),
-    DEBIAN("Debian", "debian", "debian");
+    LEAP("openSUSE Leap", "Leap", "opensuse",
+            oneOf("15.2", "15.3", "15.4", "15.5")),
+    LEAP_MICRO("openSUSELeap Micro", "openSUE Leap Micro", "opensuse",
+            oneOf("5.2", "5.3", "5.4", "5.5")),
+    SUSE_LINUX_ENTERPRISE_SERVER("SUSE Linux Enterprise Server", "SLES", "suse",
+            oneOf("11", "12", "15")),
+    SUSE_LINUX_ENTERPRISE_DESKTOP("SUSE Linux Enterprise Desktop", "SLED", "suse",
+            oneOf("10", "11", "12", "15")),
+    SUSE_LINUX_ENTERPRISE_MICRO("SUSE Linux Enterprise Micro", "SLE Micro", "suse",
+            oneOf("5.0", "5.1", "5.2", "5.3", "5.4", "5.5")),
```

Comment:
SL Micro 6.0 is will be released in coming weeks, would be good to already add entry for it.

Reply:
Done. I have added an entry to the config file too, it points to an empty URL for now.